packages/packagers/js/src/ScopeHoistingPackager.js

```diff
@@ -130,9 +131,19 @@ export class ScopeHoistingPackager {
       this.bundle.env.isLibrary ||
       this.bundle.env.outputFormat === 'commonjs'
     ) {
-      let bundles = this.bundleGraph.getReferencedBundles(this.bundle);
-      for (let b of bundles) {
-        this.externals.set(relativeBundlePath(this.bundle, b), new Map());
+      for (let b of this.bundleGraph.getReferencedBundles(this.bundle)) {
+        let entry = b.getMainEntry();
+        let symbols = new Map();
+        if (entry && !this.isAsyncBundle && entry.type === 'js') {
```

Comment:
for now async bundles in libraries are still referenced with `parcelRequire`. Changing that would require additional changes to the runtimes, but we could fix it eventually.